utils/useEtherscanUrl.ts

```diff
@@ -0,0 +1,32 @@
+import Connection from "../containers/Connection";
+
+const networkToEtherscanHostMap = (networkName: string | undefined) => {
+    switch(networkName) {
+        case "kovan":
+            return "https://kovan.etherscan.io";
+        case "ropsten":
+            return "https://ropsten.etherscan.io";
+        case "rinkeby":
+            return "https://rinkeby.etherscan.io";
+        case "goerli":
+            return "https://goerli.etherscan.io";
+        default:
+            return "https://etherscan.io"
+    }
+}
+
+export const useEtherscanUrl = (txnHash: string | null) => {
+    if (!txnHash) { 
+      return undefined; 
+    }
+    else {
+        const { network } = Connection.useContainer();
+        const etherscanHost = networkToEtherscanHostMap(network?.name);
+    
+        return `${etherscanHost}/tx/${txnHash}`;  
+    }
+};
```

Comment:
```suggestion
interface NetworkMap {
  [key: string]: string;
}

const NETWORK_MAP: NetworkMap = {
  kovan: "https://kovan.etherscan.io",
  ropsten: "https://ropsten.etherscan.io",
  rinkeby: "https://rinkeby.etherscan.io",
  goerli: "https://goerli.etherscan.io",
  mainnet: "https://etherscan.io",
};

export const useEtherscanUrl = (txnHash: string | null) => {
  const { network } = Connection.useContainer();
  if (!txnHash || !network) {
    return undefined;
  }

  const baseUrl = NETWORK_MAP[network.name] || NETWORK_MAP["mainnet"];
  return `${baseUrl}/tx/${txnHash}`;
};
```

Reasons for suggested changes:

- Refrain from creating [new functions](https://www.youtube.com/watch?v=Bks59AaHe1c) as they signal complexity and can cause [indirection](https://twitter.com/mpjme/status/814250097651355649). Here the job can be accomplished by a well-typed object literal doing the mapping.

- Hooks (including `Connection.useContainer`) should [never](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) be inside conditionals, so that means the first thing we do is we always call it at the top of our custom hook (which is what this file is).

- Also it's good to always have a return at the end. An `else-return` pattern can sometimes be a code smell as it is frequently [unnecessary](https://eslint.org/docs/rules/no-else-return).

btw, I agree that we should probably fold these guidelines into a linter.

Reply:
Nice I agree with all of these and especialy like the use of the interface `NetworkMap`